clar2wasm/tests/wasm-generation/control_flow.rs

```diff
@@ -62,3 +64,51 @@ proptest! {
         );
     }
 }
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin(val in PropValue::any()) {
+        crosscheck(
+            &format!(r#"(begin (some {val}) {val})"#),
+            Ok(Some(val.into()))
+        );
+    }
+}
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin_ok(val in PropValue::any()) {
+        crosscheck(
+            &format!(r#"(begin (some {val}) (ok {val}))"#),
+            Ok(Some(Value::Response(ResponseData { committed: true, data: Box::new(val.into()) })))
+        );
+    }
+}
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin_err(val in PropValue::any()) {
+        crosscheck(
+            &format!(r#"(begin (some {val}) (err {val}))"#),
+            Ok(Some(Value::Response(ResponseData { committed: false, data: Box::new(val.into()) })))
+        );
+    }
```

Comment:
What is the point of this test? What does it tell us that the previous test doesn’t tell already?

Reply:
See comment on `begin_ok`